### PR TITLE
seamly2d: init at 2022-08-15.0339

### PR DESCRIPTION
**Pull request by WhittlesJr**
###### Description of changes

Adds Seamly2D and its companion application, SeamlyMe. This is a vector-based fashion pattern drafting application. It is a fork of Valentina, which is already packaged for NixOS by @jfrankenau

This is my second PR. The first one was rebased funny and automatically pulled in a ton of reviewers. It got closed and locked.

https://seamly.net/
https://github.com/FashionFreedom/Seamly2D

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/seamly2d/default.nix [hunk 1]
```diff
@@ -0,0 +1,95 @@
+{ stdenv, lib, qtbase, wrapQtAppsHook, fetchFromGitHub,
+  addOpenGLRunpath, poppler_utils, qtxmlpatterns, qtsvg, mesa, gcc, xvfb-run,
+  fontconfig, freetype, xorg, ccache, qmake, python3, qttools, git
+}:
+with lib;
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/seamly2d/default.nix:5]: Please don't use with over the entire filr

File: pkgs/applications/graphics/seamly2d/default.nix [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    repo = "Seamly2D";
+    rev = "v${version}";
+    sha256 = "13jxkg84jfz8g52zwhh5jvi23wryzkavwbsfalzr9m04blj5xnik";
+  };
+
+  buildInputs = [
+    git
+    qtPython
+    qtbase
+    addOpenGLRunpath
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/seamly2d/default.nix:24]: Goes to nativeBuildInputs

File: pkgs/applications/graphics/seamly2d/default.nix [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    poppler_utils
+    qtxmlpatterns
+    qtsvg
+    mesa
+    gcc
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/seamly2d/default.nix:29]: Part of stdenv

File: pkgs/applications/graphics/seamly2d/default.nix [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    xvfb-run
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/seamly2d/default.nix:30]: Native

File: pkgs/applications/graphics/seamly2d/default.nix [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    fontconfig
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/seamly2d/default.nix:31]: Also nativeBuildInputs

File: pkgs/applications/graphics/seamly2d/default.nix [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    freetype
+    xorg.libXi
+    xorg.libXrender
+    xorg.libxcb
+    ccache
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/seamly2d/default.nix:36]: Also native and probably not useful for nix. I would suggest to remove it.

File: pkgs/applications/graphics/seamly2d/default.nix [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    "PREFIX_LIB=/lib"
+    "Seamly2D.pro"
+    "-r"
+    "CONFIG+=noDebugSymbols"
+    "CONFIG+=no_ccache"
+  ];
+
+  installFlags = [ "INSTALL_ROOT=$(out)" ];
+
+  enableParallelBuilding = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/seamly2d/default.nix:72]: Already set by qmake

File: pkgs/applications/graphics/seamly2d/default.nix [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+  postInstall = ''
+    mv $out/usr/share $out/
+    rmdir $out/usr
+
+    mv $out/share/seamly2d/* $out/share/.
+    rmdir $out/share/seamly2d
+
+    mkdir -p $out/share/man/man1
+    gzip -9c dist/debian/seamly2d.1 > $out/share/man/man1/seamly2d.1.gz
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/seamly2d/default.nix:82]: Automatically done in fixupPhase

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by WhittlesJr** [comment]
@SuperSandro2000 , thank you very much for the review. I apologize for the late followup. I have implemented the changes you requested in a separate commit. If you approve, I'll squash it down.

**Pull request merged**